### DISCUSSION TITLE
Can I select from database only those fields that were requested ?

### DISCUSSION THREAD

**Discussion by detwait** [TriPSs/nestjs-query#276] [General]
During logging sql under the hood I noticed that each query select all fields regardless of which ones were requested in graphql query.
Am I doing something wrong ? Or was it designed that way ?

**Comment by TriPSs** [comment]
This is indeed correct.

**Reply by detwait** [reply]
> This is indeed correct.

Why ?
Is there some blocker or tech challenge ?

**Reply by TriPSs** [reply]
We recently only added look ahead so we never had the info on which fields where selected, beside that this is normal TypeORM behaviour, also not quite sure if the performance improvement will be worth the effort plus these services can also be used without GraphQL so then we don't have the info again.

Edit: the return types are also the entities, meaning that the user now also expects the full entity to be returned (maybe you need some fields GraphqQL does not select to determine a other one).
